### PR TITLE
Yet some more uniquifying of implicit names

### PR DESCRIPTION
**Pull request by ceedubs**
I think this is the last of the implicit names for #1061.

@kailuowang would you be willing to review this?

### CODE REVIEW

File: kernel-laws/src/main/scala/cats/kernel/laws/package.scala [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
 import Prop.{False, Proof, Result}
 
 package object laws {
 
-  implicit def nonEmptyPredicate[A](implicit ev: Eq[A], A: Monoid[A]): Predicate[A] =
-    new Predicate[A] {
-      def apply(a: A) = ev.neqv(a, A.empty)
-    }
-
   lazy val proved = Prop(Result(status = Proof))
```
Comment (kailuowang) [inline kernel-laws/src/main/scala/cats/kernel/laws/package.scala]: Just curious what makes this one redundant?

Comment (ceedubs) [inline kernel-laws/src/main/scala/cats/kernel/laws/package.scala]: Oh good catch I meant to ask people about this.

@non @johnynek: this seemed a little suspicious to me. I can understand the parameters being implicit, but why was the `def` itself implicit? It wasn't actually used in cats and didn't appear to be in algebra, so I removed it. Is this something that should be kept around?

Comment (non) [inline kernel-laws/src/main/scala/cats/kernel/laws/package.scala]: Let's remove it. I'm not sure what I was thinking.

**Comment by non** [comment]
👍

**Comment by kailuowang** [comment]
👍

**Pull request merged**